### PR TITLE
Fixes: #1459

### PR DESCRIPTION
**Pull request by ubajze**
### Fixes: #1459
The PR adds the `annotate` method to clusters in `ClusterTypeView` and `ClusterGroupView`.

### CODE REVIEW

File: nautobot/virtualization/views.py [hunk 1]
```diff
@@ -38,6 +38,9 @@ def get_extra_context(self, request, instance):
             Cluster.objects.restrict(request.user, "view")
             .filter(type=instance)
             .prefetch_related("group", "site", "tenant")
+        ).annotate(
+            device_count=count_related(Device, 'cluster'),
+            vm_count=count_related(VirtualMachine, 'cluster')
```
Comment (glennmatthews) [inline nautobot/virtualization/views.py:43]:
```suggestion
            vm_count=count_related(VirtualMachine, 'cluster'),
```

File: nautobot/virtualization/views.py [hunk 2]
```diff
@@ -94,6 +97,9 @@ def get_extra_context(self, request, instance):
             Cluster.objects.restrict(request.user, "view")
             .filter(group=instance)
             .prefetch_related("type", "site", "tenant")
+        ).annotate(
+            device_count=count_related(Device, 'cluster'),
+            vm_count=count_related(VirtualMachine, 'cluster')
```
Comment (glennmatthews) [inline nautobot/virtualization/views.py:102]:
```suggestion
            vm_count=count_related(VirtualMachine, 'cluster'),
```

**Review by glennmatthews — COMMENTED** [review]
Looks good - suggested changes are to make Black happy.

File: nautobot/virtualization/views.py [hunk 3]
```diff
@@ -38,6 +38,9 @@ def get_extra_context(self, request, instance):
             Cluster.objects.restrict(request.user, "view")
             .filter(type=instance)
             .prefetch_related("group", "site", "tenant")
+        ).annotate(
+            device_count=count_related(Device, 'cluster'),
+            vm_count=count_related(VirtualMachine, 'cluster'),
```
Comment (glennmatthews) [inline nautobot/virtualization/views.py:43]:
```suggestion
            device_count=count_related(Device, "cluster"),
            vm_count=count_related(VirtualMachine, "cluster"),
```

File: nautobot/virtualization/views.py [hunk 4]
```diff
@@ -94,6 +97,9 @@ def get_extra_context(self, request, instance):
             Cluster.objects.restrict(request.user, "view")
             .filter(group=instance)
             .prefetch_related("type", "site", "tenant")
+        ).annotate(
+            device_count=count_related(Device, 'cluster'),
+            vm_count=count_related(VirtualMachine, 'cluster'),
```
Comment (glennmatthews) [inline nautobot/virtualization/views.py:102]:
```suggestion
            device_count=count_related(Device, "cluster"),
            vm_count=count_related(VirtualMachine, "cluster"),
```

**Review by glennmatthews — APPROVED** [review]

**Comment by ubajze** [comment]
@glennmatthews thx

**Pull request merged**